File: _FACUL/LP/LP_Go/string_search.go
package main

import (
	"bufio"
	f "fmt"
	"os"
	s "strings"
)

var str, busca, temp string
var vezes int = 0

func main() {
	f.Print("Digite uma frase: ")
	read := bufio.NewReader(os.Stdin)
	str, _ = read.ReadString('\n')
	f.Print("Qual a busca: ")
	busca, _ = read.ReadString('\n')
	str = s.TrimSpace(str)
	busca = s.TrimSpace(busca)

	for i := 0; i < len(str); i++ {
		temp += string(str[i])
		if temp == busca {
			vezes++
		}
		if len(temp) == len(busca) {
			temp = ""
			i -= (len(busca) - 1)
		}
	}
	f.Printf("A palavra '%s' aparece %d vezes", busca, vezes)
}

File: _FACUL/LP/LP_C/string_search.c
#include <stdio.h>
#include <string.h>

int main(void){
  char word[100], phrase[100], temp[100];
  int i, j = 0, vezes = 0;

  printf("Digite uma frase: ");
  gets(phrase);
  printf("Digite uma palavra: ");
  gets(word);

  for(i = 0; i < strlen(phrase); i++){
    temp[j] = phrase[i];
    j++;
    temp[j] = '\0';
    if(!strcmp(temp, word)){
      j = 0;
      vezes++;
    }
    if(strlen(temp) == (strlen(word))){
      j = 0; 
      i-= (strlen(word) - 1);
    }
  }
  printf("\nA palavra %s aparece %d vezes", word, vezes);
  return 0;
}